components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/display/DisplayToolbar.kt

```diff
@@ -176,7 +185,13 @@ internal class DisplayToolbar(
 
     private val defaultColor = ContextCompat.getColor(context, R.color.photonWhite)
 
+    private var currentSiteSecurity = SiteSecurity.INSECURE
+
     internal var securityIconColor = Pair(defaultColor, defaultColor)
+        set(value) {
+            field = value
+            setSiteSecurity(currentSiteSecurity)
+        }
```

Comment:
How is this change related? What's the background?

We try really hard to not have any state here and keep that in the `Session` only.

Reply:
I initially filed [a separate issue](https://github.com/mozilla-mobile/android-components/issues/2620) for that change, but wasn't sure if that was necessary. I'm happy to handle it in a separate PR :)